tls/s2n_server_hello.c

```diff
@@ -182,7 +186,7 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
 
     /* If this is a HelloRetryRequest, we don't process the ServerHello.
      * Instead we proceed with retry logic. */
-    if (s2n_hello_retry_validate(conn) == S2N_SUCCESS) {
+    if ((s2n_hello_retry_validate(conn) == S2N_SUCCESS) && (conn->actual_protocol_version >= S2N_TLS13)) {
```

Comment:
Shouldn't we fail if we receive a retry request when the actual_protocol_version isn't TLS1.3? Just ignoring it and treating it as a server_hello sounds wrong. If you do fail on TLS1.3, it should probably happen in s2n_server_hello_retry_recv.

Reply:
I can make this check within s2n_server_hello_retry_recv